Source/Initialization/WarpXInitData.cpp

```diff
@@ -1228,3 +1289,152 @@ void WarpX::CheckKnownIssues()
                 ablastr::warn_manager::WarnPriority::low);
         }
 }
+
+#if defined(WARPX_USE_OPENPMD) && !defined(WARPX_DIM_1D_Z)
+void
+WarpX::ReadExternalFieldFromFile (
+       std::string read_fields_from_path, MultiFab* mf,
+       std::string F_name, std::string F_component)
+{
+    // Get WarpX domain info
+    auto& warpx = WarpX::GetInstance();
+    amrex::Geometry const& geom0 = warpx.Geom(0);
+    const amrex::RealBox& real_box = geom0.ProbDomain();
+    const auto dx = geom0.CellSizeArray();
+    amrex::IntVect nodal_flag = mf->ixType().toIntVect();
+
+    // Read external field openPMD data
+    auto series = openPMD::Series(read_fields_from_path, openPMD::Access::READ_ONLY);
+    auto iseries = series.iterations.begin()->second;
+    auto F = iseries.meshes[F_name];
+    auto offset = F.gridGlobalOffset();
+    auto offset0 = offset[0];
+    auto offset1 = offset[1];
+#if defined(WARPX_DIM_3D)
+    auto offset2 = offset[2];
+#endif
```

Comment:
Would it be possible to use the explicit types instead of `auto`, at least for the easy cases? So that code reades can understand immediately if they are dealing with, say, an integer or floating point offset, for example.